### PR TITLE
Nunavut Age Amount Credit

### LINKED ISSUE #275: Nunavut Age Amount Credit
Reference: https://www.nunavutlegislation.ca/en/consolidated-law/income-tax-act-consolidation p-13

### PR DESCRIPTION
**Pull request by runyao-yin**
fixes #275

### CODE REVIEW

File: changelog_entry.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Add nunavut age amount credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Nunavut age amount credit.
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Nunavut provides this maximum age amount credit for eligible people
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:1]:
```suggestion
description: Nunavut provides this maximum age amount credit for eligible people.
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2022-07-01: 7_500
+metadata:
+  unit: currency-CAD
+  label: Nunavut age amount credit maximum amount
+  reference:
+  - title: Consolidation of Income Tax Act, Part I Division A 2.2
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:8]:
```suggestion
  - title: Consolidation of Income Tax Act, Part I Division B 2.2
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:8]: and others

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+description: Nunavut allows people who are over this age to receive the age amount credit. 
+values:
+  2023-01-01: 65
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml:3]:
```suggestion
  2002-01-01: 65
```

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]
Mostly format

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml (hunk 4 above)
Comment (runyao-yin) [reply policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml:3]: Wondering why we are using 2002 here?

Comment (MaxGhenis) [reply policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml:3]: The title of the law is `Age credit for 2002 and subsequent taxation years`

**Comment by runyao-yin** [comment]
Accept changes

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+description: Nunavut provides this maximum age amount credit for eligible people.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:1]:
```suggestion
description: Nunavut provides this maximum age amount credit for eligible filers.
```

**Comment by runyao-yin** [comment]
Accept changes

**Comment by runyao-yin** [comment]
Changes added

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+class nu_age_amount_credit_eligible_person(Variable):
+    value_type = bool
+    entity = Person
+    label = "Nunvaut age amount credit eligible person"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NU
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        return (
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:13]:
```suggestion
        p = parameters(
                period
            ).gov.provinces.nu.tax.income.credits.age_amount
        return age >= p.eligible_age
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 7]
```diff
@@ -0,0 +1,18 @@
+from policyengine_canada.model_api import *
+
+
+class nu_age_amount_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "Nunvaut age amount credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = ProvinceCode.NU
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:10]: And remove from formula
```suggestion
    defined_for = "nu_age_amount_credit_eligible_person"
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Comment by runyao-yin** [comment]
Accept changes. Wondering what i need to do to pass the test again?

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    value_type = float
+    entity = Person
+    label = "Nunvaut age amount credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "nu_age_amount_credit_eligible_person"
+
+    def formula(person, period, parameters):
+        income = person("total_individual_pre_tax_income", period)
+        eligible = person("nu_age_amount_credit_eligible_person", period)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:14]:
```suggestion
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+        p = parameters(period).gov.provinces.nu.tax.income.credits.age_amount
+        phase_out_amount = p.phase_out_rate.calc(income)
+        max_amount = p.amount
+        return eligible * max_(max_amount - phase_out_amount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:18]:
```suggestion
        return max_(max_amount - phase_out_amount, 0)
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+    label = "Nunvaut age amount credit eligible person"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NU
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        p = parameters(
+                period
+            ).gov.provinces.nu.tax.income.credits.age_amount
+        return age >= p.eligible_age
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:16]: return only one of the two

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml [hunk 11]
```diff
@@ -0,0 +1,9 @@
+description: Nunavut allows people who are over this age to receive the age amount credit.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/eligible_age.yaml:1]:
```suggestion
description: Nunavut allow filers at or over this age to receive the age amount credit. 
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 12]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  2022-07-01: 10_764
+metadata:
+  unit: currency-CAD
+  label: Nunavut age amount credit maximum amount
+  reference:
+  - title: Consolidation of Income Tax Act, Part I Division B Subdivision c 2.2
+    href: https://www.nunavutlegislation.ca/en/consolidated-law/income-tax-act-consolidation
+  - title: Worksheet NU428
+    href: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5014-d/5014-d-22e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:11]: add page number as .pdf#page=x

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 13]
```diff
@@ -0,0 +1,12 @@
+description: Nunavut provides this maximum age amount credit for eligible people.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:1]:
```suggestion
description: Nunavut provides this maximum age amount credit for eligible filers.
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 14]
```diff
@@ -0,0 +1,12 @@
...
+values:
+  2022-07-01: 10_764
+metadata:
+  unit: currency-CAD
+  label: Nunavut age amount credit maximum amount
+  reference:
+  - title: Consolidation of Income Tax Act, Part I Division B Subdivision c 2.2
+    href: https://www.nunavutlegislation.ca/en/consolidated-law/income-tax-act-consolidation
+  - title: Worksheet NU428 Page 1
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:10]:
```suggestion
  - title: Worksheet NU428 
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml [hunk 15]
```diff
@@ -0,0 +1,12 @@
+description: Nunavut provides this maximum age amount credit for eligible people.
+values:
+  2022-07-01: 10_764
+metadata:
+  unit: currency-CAD
+  label: Nunavut age amount credit maximum amount
+  reference:
+  - title: Consolidation of Income Tax Act, Part I Division B Subdivision c 2.2
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:8]: comment: "# Law specifies amount that is not adjusted for inflation"

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml [hunk 16]
```diff
@@ -0,0 +1,20 @@
+description: Nunavut age amount credit's maximum amount is phased out at this reduction rate.
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml:1]:
```suggestion
description: Nunavut phases the age amount credit out at this rate, based on income.
```

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+  - threshold:
+      2022-07-01: 39_826
+    rate:
+      2022-07-01: 0.15
+
+metadata:
+  type: marginal_rate
+  rate_unit: /1
+  threshold_unit: currency-CAD
+  label: Nunavut age amount credit maximum amount reduction
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml:16]:
```suggestion
  label: Nunavut age amount credit reduction rate
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 18]
```diff
@@ -0,0 +1,17 @@
...
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "nu_age_amount_credit_eligible_person"
+
+    def formula(person, period, parameters):
+        income = person("total_individual_pre_tax_income", period)
+        p = parameters(period).gov.provinces.nu.tax.income.credits.age_amount
+        phase_out_amount = p.phase_out_rate.calc(income)
+        max_amount = p.amount
+        return eligible * max_(max_amount - phase_out_amount, 0)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:17]:
```suggestion
        return max_(max_amount - phase_out_amount, 0)
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 19]
```diff
@@ -0,0 +1,17 @@
+from policyengine_canada.model_api import *
+
+
+class nu_age_amount_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "Nunvaut age amount credit"
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:7]:
```suggestion
    label = "Nunavut age amount credit"
```
And elsewhere

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml (hunk 15 above)
Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/amount.yaml:8]: let me know when u commit it

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py [hunk 20]
```diff
@@ -0,0 +1,16 @@
+from policyengine_canada.model_api import *
+
+
+class nu_age_amount_credit_eligible_person(Variable):
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:4]: do we need `_person` in this?

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 21]
```diff
@@ -0,0 +1,17 @@
...
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "nu_age_amount_credit_eligible_person"
+
+    def formula(person, period, parameters):
+        income = person("total_individual_pre_tax_income", period)
+        p = parameters(period).gov.provinces.nu.tax.income.credits.age_amount
+        phase_out_amount = p.phase_out_rate.calc(income)
+        max_amount = p.amount
+        return max_(max_amount - phase_out_amount, 0)
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:17]: can skip `max_amount`
```suggestion
        return max_(p.amount - phase_out_amount, 0)
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py (hunk 20 above)
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:4]:
```suggestion
class nu_age_amount_credit_eligible(Variable):
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py [hunk 22]
```diff
@@ -0,0 +1,16 @@
...
+    value_type = bool
+    entity = Person
+    label = "Nunvaut age amount credit eligible person"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:7]:
```suggestion
    label = "Eligible for the Nunavut age amount credit"
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py [hunk 23]
```diff
@@ -0,0 +1,16 @@
+from policyengine_canada.model_api import *
+
+
+class nu_age_amount_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "Nunavut age amount credit"
+    unit = CAD
+    definition_period = YEAR
+    defined_for = "nu_age_amount_credit_eligible_person"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit.py:10]:
```suggestion
    defined_for = "nu_age_amount_credit_eligible"
```

File: policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py [hunk 24]
```diff
@@ -0,0 +1,16 @@
...
+    value_type = bool
+    entity = Person
+    label = "Eligible for the Nunavut age amount credit"
+    definition_period = YEAR
+    defined_for = ProvinceCode.NU
+
+    def formula(person, period, parameters):
+        age = person("age", period)
+        p = parameters(
+                period
```
Comment (MaxGhenis) [inline policyengine_canada/variables/gov/provinces/nu/tax/income/credits/age_amount/nu_age_amount_credit_eligible_person.py:14]: fix indent

File: policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml [hunk 25]
```diff
@@ -0,0 +1,20 @@
...
+    rate:
+      2022-07-01: 0.15
+
+metadata:
+  type: marginal_rate
+  rate_unit: /1
+  threshold_unit: currency-CAD
+  label: Nunavut age amount credit reduction rate
+  reference:
+  - title: Consolidation of Income Tax Act, Part I Division B Subdivision c 2.2
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml:18]: tax form reference?

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nu/tax/income/credits/age_amount/phase_out_rate.yaml:18]: all yaml files

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**